src/runmode-af-xdp.c

```diff
@@ -0,0 +1,428 @@
+                1) {
+            if (conf_val_int) {
+                aconf->busy_poll_time = conf_val_int;
+            }
+        }
+
+        if (ConfGetChildValueIntWithDefault(
+                    if_root, if_default, "busy-poll-budget", &conf_val_int) == 1) {
+            if (conf_val_int) {
+                aconf->busy_poll_budget = conf_val_int;
+            }
+        }
+
+        /* 0 value is valid for these Linux tunable's */
+        if (ConfGetChildValueIntWithDefault(
+                    if_root, if_default, "gro-flush-timeout", &conf_val_int) == 1) {
+            aconf->gro_flush_timeout = conf_val_int;
+        }
+
+        if (ConfGetChildValueIntWithDefault(
+                    if_root, if_default, "napi-defer-hard-irq", &conf_val_int) == 1) {
+            aconf->napi_defer_hard_irqs = conf_val_int;
+        }
+    }
+#endif
+
+finalize:
+    return aconf;
+}
+
+static int AFXDPConfigGetThreadsCount(void *conf)
+{
+    if (conf == NULL)
+        FatalError(SC_ERR_AFXDP_CONF, "Configuration file is NULL");
+
+    AFXDPIfaceConfig *afxdp_conf = (AFXDPIfaceConfig *)conf;
+    return afxdp_conf->threads;
+}
+
+int AFXDPRunModeIsIPS()
+{
+    return 0;
+}
+
+#endif /* HAVE_AF_XDP */
+
+/**
+ * \brief Single thread version of the AF_XDP processing.
+ */
+int RunModeIdsAFXDPSingle(void)
+{
+    SCEnter();
+
+#ifdef HAVE_AF_XDP
+    int ret;
+    const char *live_dev = NULL;
+
+    RunModeInitialize();
+    TimeModeSetLive();
+
+    (void)ConfGet("af-xdp.live-interface", &live_dev);
+
+    if (AFXDPQueueProtectionInit() != TM_ECODE_OK) {
+        FatalError(SC_ERR_FATAL, "Unable to init AF_XDP queue protection.");
+    }
+
+    ret = RunModeSetLiveCaptureSingle(ParseAFXDPConfig, AFXDPConfigGetThreadsCount, "ReceiveAFXDP",
+            "DecodeAFXDP", thread_name_single, live_dev);
+    if (ret != 0) {
+        FatalError(SC_ERR_FATAL, "Unable to start runmode");
+    }
+
+    SCLogDebug("RunModeIdsAFXDPSingle initialised");
+
+#endif /* HAVE_AF_XDP */
+    SCReturnInt(0);
+}
+
+/**
+ * \brief Workers version of the AF_XDP processing.
+ *
+ * Start N threads with each thread doing all the work.
+ *
+ */
+int RunModeIdsAFXDPWorkers(void)
+{
+    SCEnter();
+
+#ifdef HAVE_AF_XDP
+    int ret;
+    const char *live_dev = NULL;
+
+    RunModeInitialize();
+    TimeModeSetLive();
+
+    (void)ConfGet("af-xdp.live-interface", &live_dev);
+
+    if (AFXDPQueueProtectionInit() != TM_ECODE_OK) {
+        FatalError(SC_ERR_FATAL, "Unable to init AF_XDP queue protection.");
+    }
+
+    ret = RunModeSetLiveCaptureWorkers(ParseAFXDPConfig, AFXDPConfigGetThreadsCount, "ReceiveAFXDP",
+            "DecodeAFXDP", thread_name_workers, live_dev);
+    if (ret != 0) {
+        FatalError(SC_ERR_FATAL, "Unable to start runmode");
+    }
+
+    SCLogDebug("RunModeIdsAFXDPWorkers initialised");
+
+#endif /* HAVE_AF_XDP */
+    SCReturnInt(0);
+}
+
+int RunModeIdsAFXDPAutoFp(void)
```

Comment:
curious if you see a use case for autofp. Not hard to leave it out. Hard to remove it later :)

Reply:
Honestly, no. I just wasn't sure if its best to have it there and be told to remove it :) Can quite gladly leave it out.

Reply:
ok lets remove it then, can also add it later if there is strong demand, which seems doubtful as of now